### PR TITLE
ListReorder quiz functional

### PR DESCRIPTION
**Pull request by PimCoumans**
Read ‘em and weep!

### CODE REVIEW

File: src/Components/Quiz/ListReorder.js [hunk 1]
```diff
@@ -9,37 +11,46 @@ const questions_drag_and_drop = [
...
     ); // Houd de items bij voor elke vraag
+    const [allAnswered, setAllAnswered] = useState(false);
+    const [errors, setErrors] = useState([]);
+    const { score, setScore } = useContext(ScoreContext);
+    const navigate = useNavigate();
 
-    const handleAnswerSelection = (questionIndex, selectedAnswer) => {
-        const updatedAnswers = [...answers];
-        updatedAnswers[questionIndex] = selectedAnswer;
+    const handleAnswerSelection = (questionIndex, order) => {
```
Comment (Nynkek) [inline src/Components/Quiz/ListReorder.js:73]: de answers moeten ook nog in de local storage opgeslagen worden zodat je na een refresh de antwoorden houdt

**Review by Nynkek — APPROVED** [review]
lekker bezig knappie

**Comment by PimCoumans** [comment]
Gebruik nu `localStorage` voor lijst volgorde en antwoorden! 🥳

**Comment by PimCoumans** [comment]
Rebased nu ook, klaar om te mergen

**Review by Nynkek — COMMENTED** [review]
Code Improvements
Use Functional State Updates: In handleAnswerSelection, consider using the functional form of setAnswers to ensure you work with the latest state:
javascript
Copy code
setAnswers(prevAnswers => {
    const updatedAnswers = [...prevAnswers];
    updatedAnswers[questionIndex] = order.map(answer => options.indexOf(answer));
    return updatedAnswers;
});
Avoid Direct State Mutation: When you update state like const updatedAnswers = answers;, you're mutating the state directly. Always create a new copy:
javascript
Copy code
const updatedAnswers = [...answers];
Improve Local Storage Functions: You could create a single function to handle both saving and loading from local storage. This reduces repetition:
javascript
Copy code
const localStorageHandler = (key, data) => {
    if (data) {
        localStorage.setItem(key, JSON.stringify(data));
    } else {
        return JSON.parse(localStorage.getItem(key)) || [];
    }
};
Simplify arraysEqual: You can use Array.prototype.every for a cleaner check:
javascript
Copy code
const arraysEqual = (left, right) => 
    left.length === right.length && left.every((value, index) => value === right[index]);
Conditional Rendering for Error Messages: You can simplify error message rendering by checking if the error exists:
javascript
Copy code
{errors[questionIndex] && <p className="error">Dit antwoord is fout, probeer opnieuw!</p>}
Styling Classes: Consider using a class library or utility classes (like Tailwind CSS) for a more manageable and responsive design.

**Pull request merged**